SpatialGDK/Source/SpatialGDK/Public/SpatialView/ConnectionHandlers/SpatialOSConnectionHandler.h

```diff
@@ -0,0 +1,190 @@
+			}
+		}
+
+		return Ops;
+	}
+
+	void SendMessages(TUniquePtr<MessagesToSend> Messages) override
+	{
+		const Worker_UpdateParameters UpdateParams = {0 /*loopback*/};
+		const Worker_CommandParameters CommandParams = {0 /*allow_short_circuit*/};
+		for (auto& Message : Messages->ComponentMessages)
+		{
+			switch (Message.GetType()) {
+			case OutgoingComponentMessage::ADD:
+			{
+				Worker_ComponentData Data = {nullptr, Message.ComponentId,
+					MoveTemp(Message).ReleaseComponentAdded().Release(), nullptr};
+				Worker_Connection_SendAddComponent(Connection.Get(), Message.EntityId, &Data, &UpdateParams);
+				break;
+			}
+			case OutgoingComponentMessage::UPDATE:
+			{
+				Worker_ComponentUpdate Update = {nullptr, Message.ComponentId,
+					MoveTemp(Message).ReleaseComponentUpdate().Release(), nullptr};
+				Worker_Connection_SendComponentUpdate(Connection.Get(), Message.EntityId, &Update, &UpdateParams);
+				break;
+			}
+			case OutgoingComponentMessage::REMOVE:
+			{
+				Worker_Connection_SendRemoveComponent(Connection.Get(),
+					Message.EntityId, Message.ComponentId, &UpdateParams);
+				break;
+			}
+			default:
+				check(false);
+				break;
+			}
+		}
+
+		for (auto& Request : Messages->CreateEntityRequests)
+		{
+			TArray<Worker_ComponentData> Components;
+			Components.Reserve(Request.EntityComponents.Num());
+			for (ComponentData& Component :  Request.EntityComponents)
+			{
+				Components.Push(Worker_ComponentData{nullptr, Component.GetComponentId(), MoveTemp(Component).Release(),
+					nullptr});
+			}
+			Worker_EntityId* EntityId = Request.EntityId.IsSet() ? Request.EntityId.GetValue() : nullptr;
+			const uint32* Timeout = Request.TimeoutMillis.IsSet() ? Request.TimeoutMillis.GetValue() : nullptr;
+			const Worker_RequestId Id = Worker_Connection_SendCreateEntityRequest(Connection.Get(), Components.Num(),
+				Components.GetData(), EntityId, Timeout);
+			InternalToUserRequestId[Id] = Request.RequestId;
+		}
+
+		for (auto& Request : Messages->DeleteEntityRequests)
+		{
+			const uint32* Timeout = Request.TimeoutMillis.IsSet() ? Request.TimeoutMillis.GetValue() : nullptr;
+			const Worker_RequestId Id = Worker_Connection_SendDeleteEntityRequest(Connection.Get(), Request.EntityId,
+				Timeout);
+			InternalToUserRequestId[Id] = Request.RequestId;
+		}
+
+		for (auto& Request : Messages->EntityQueryRequests)
+		{
+			const uint32* Timeout = Request.TimeoutMillis.IsSet() ? Request.TimeoutMillis.GetValue() : nullptr;
+			Worker_EntityQuery Query = Request.Query.GetWorkerQuery();
+			const Worker_RequestId Id = Worker_Connection_SendEntityQueryRequest(Connection.Get(), &Query, Timeout);
+			InternalToUserRequestId[Id] = Request.RequestId;
+		}
+
+		for (auto& Request : Messages->EntityCommandRequests)
+		{
+			const uint32* Timeout = Request.TimeoutMillis.IsSet() ? Request.TimeoutMillis.GetValue() : nullptr;
+			Worker_CommandRequest r = {nullptr, Request.Request.GetComponentId(), Request.Request.GetCommandIndex(),
```

Comment:
Can we change this variable name from "r"

Reply:
We can but it lives for one line because it was too long to insert it into the line below (where it should really be).
I'm not sure the name of this affects readability.